cEP-0005.md

```diff
@@ -0,0 +1,157 @@
+# coala Configuration
+
+| Metadata     |                                               |
+| ------------ |-----------------------------------------------|
+| cEP          | 5                                             |
+| Version      | 0.1                                           |
+| Title        | coala Configuration                           |
+| Authors      | Lasse Schuirmann <lasse.schuirmann@gmail.com> |
+| Status       | Proposed                                      |
+| Type         | Feature                                       |
+
+## Problem
+
+One main pain point for our users is that coala is hard to configure. It tries
+to provide full options to a full set of tools. At the same time we try to
+provide an abstraction over the tools by e.g. giving the same setting names to
+similar bears across languages so the user can reuse the same configuration for
+other languages too.
+
+By integrating lots of different existing tools with coala we have gained a
+huge set of possible analysis but it makes it very hard to understand and
+configure without knowledge about a particular tool. Also the conflict of bears
+versus linter bears that effectively offer multiple "checks" or sets of
+analyses rises: it is confusing to have to use multiple bears for one language
+for a set of analyses that is provided by only one bear for another language.
+
+Other problems of the configuration file are, that some duplication is
+involved, especially when dealing with settings that are used for file
+collection.
+
+## Proposal
+
+One important aspect of coala and its usability is that the configuration of a
```

Comment:
imo we should provide a cool API so that when a user wants to write a bear, he starts with a default template and then has to choose from that writing_bears api  whatever arguments or functions he needs to make that bear run at its full potential